Type inference broken in 6.9.0

## Description

(From internal conversation with @iocanel and @metacosm)


PR #5138, in scope of #5135, provided changes to make Resources `Editable`.

While this improvement allows for one line edit operations similar to what Lombok's `toBuilder` does, it has somehow broken type inference in some scenarios.

One such scenario is where you have elements that are parameterized as such: `A<R extends HasMetadata, P>`. If you have several such objects which all have the same value for `P` but different values for `R` and you want to process them as a list. 

Before 6.9.0, the list was correctly inferred to be a list containing elements of type `A<? extends HasMetadata, P>`.
With 6.9.0, the list is now inferred to contain elements of type `A<? extends Editable<? extends BaseFluent<? extends BaseFluent<?>>>, P>`.

With type erasure, this results in hard-to-diagnose errors because they occur outside of the user's code, similar to:

```java
accept(java.util.List<java.util.Map.Entry<java.lang.String,java.lang.Object>>,java.lang.String,io.fabric8.kubernetes.api.builder.Visitor...) in io.fabric8.kubernetes.api.builder.BaseFluent cannot implement accept(java.util.List<java.util.Map.Entry<java.lang.String,java.lang.Object>>,java.lang.String,io.fabric8.kubernetes.api.builder.Visitor...) in io.fabric8.kubernetes.api.builder.Visitable
[ERROR]   return type capture#1 of ? is not compatible with capture#2 of ?
```

You can see such an error here: https://github.com/operator-framework/java-operator-sdk/actions/runs/6456675029/job/17526638395#step:5:2605
If you look at the code, you'll see that none of the classes or method mentioned in the error are used, which makes for a very unpleasant diagnosing experience.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 
 #### Bugs
 * Fix #5501: [crd-generator] Fix fallback value of `Default` annotation in presence of multiple accessors
+* Fix #5522: type inference fixed by bumping Sundrio to 0.101.2 (see https://github.com/sundrio/sundrio/pull/431)
 
 #### Improvements
 

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/TypeInferenceTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/TypeInferenceTest.java
@@ -27,15 +27,13 @@ import java.util.Optional;
 
 class TypeInferenceTest {
 
-  // Commented-out due to compilation failure
-  // TODO: Enable after #5522 is fixed
-  //  @Test
-  //  void typeInferenceWithLambda() {
-  //    final ConfigMapDependentResource cdr = new ConfigMapDependentResource();
-  //    final DeploymentDependentResource ddr = new DeploymentDependentResource();
-  //    Arrays.asList(cdr, ddr).forEach(dr -> dr.reconcile());
-  //    Assertions.assertTrue(true);
-  //  }
+  @Test
+  void typeInferenceWithLambda() {
+    final ConfigMapDependentResource cdr = new ConfigMapDependentResource();
+    final DeploymentDependentResource ddr = new DeploymentDependentResource();
+    Arrays.asList(cdr, ddr).forEach(dr -> dr.reconcile());
+    Assertions.assertTrue(true);
+  }
 
   @Test
   void typeInferenceWithMethodReference() {

--- a/pom.xml
+++ b/pom.xml
@@ -90,7 +90,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
     <!-- Core versions -->
-    <sundrio.version>0.101.1</sundrio.version>
+    <sundrio.version>0.101.2</sundrio.version>
     <okhttp.version>3.12.12</okhttp.version>
     <okhttp.bundle.version>3.12.1_1</okhttp.bundle.version>
     <okio.version>1.15.0</okio.version>

